crates/transaction-pool/src/validate.rs

```diff
@@ -80,6 +81,48 @@ pub trait TransactionValidator: Send + Sync {
     }
 }
 
+pub struct EthTransactionValidator<Client: AccountProvider> {
+    /// This type fetches account info from the db
+    client: Client,
+    /// whether shanghai is enabled.
+    shanghai: bool,
+}
+
+#[async_trait::async_trait]
+impl<T: PoolTransaction + AccountProvider> TransactionValidator for EthTransactionValidator<T> {
+    type Transaction = T;
+
+    async fn validate_transaction(
+        &self,
+        origin: TransactionOrigin,
+        transaction: Self::Transaction,
+    ) -> TransactionValidationOutcome<Self::Transaction> {
+        
+        // TODO : Checks for chainid
+
+        // TODO : Checks for gaslimit
+
+        let account = self.client.basic_account(transaction.sender())?.unwrap_or_default();
```

Comment:
this is a good point, that this can fail.

I need to think about what to do here, but I guess we need to turn the response into a Result